packages/shared/src/components/post/PostActions.tsx

```diff
@@ -91,21 +91,30 @@ export function PostActions({
     }) => {
       const { entityId, type } = mutationVariables as AwardProps;
 
-      if (type !== 'POST') {
-        return;
-      }
+      if (type === 'POST') {
+        if (entityId !== post.id) {
+          return;
+        }
+
+        updatePostCache(mutationQueryClient, post.id, {
+          userState: {
+            ...post.userState,
+            awarded: true,
+          },
+          numAwards: (post.numAwards || 0) + 1,
+        });
 
-      if (entityId !== post.id) {
         return;
       }
 
-      updatePostCache(mutationQueryClient, post.id, {
-        userState: {
-          ...post.userState,
-          awarded: true,
-        },
-        numAwards: (post.numAwards || 0) + 1,
-      });
+      if (type === 'COMMENT') {
+        mutationQueryClient.invalidateQueries({
+          queryKey: generateQueryKey(RequestKey.PostComments, undefined, {
+            postId: post.id,
+          }),
+          exact: false,
+        });
```

Comment:
I invalidate in this case just because comment structure is a bit complicated and optimistic update would be a pain (with markdown and such)